download/npmExternalToInternalRedirect/npmExternalToInternalRedirect.groovy

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2014 JFrog Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.artifactory.repo.RepoPath
+import org.artifactory.repo.RepoPathFactory
+
+
+download {
+
+    altRemoteContent { repoPath ->
+        String repoKey = repoPath.getRepoKey()
+        String localNpmRepo = "npm-local";
+        RepoPath newPath = null;
+        boolean foundInLocal = false;
+        //change it to match only the external npm name
+        if (repoKey.contains("npm") && externalNpmRepo_prop.contains(repoKey)) {
```

Comment:
I will agree on this point, as it will make the plugin generic. But we wanted specific filtering based on names for now. 
Since the plugin is going to be generic i will correct and upload a new one.